File: AalsiTrader/Models/PaperPortfolio.swift
import Foundation

struct PaperPortfolio: Codable {
    var capital: Double
    var startingCapital: Double
    var availableCapital: Double
    var marginUsed: Double
    var totalPnl: Double
    var unrealizedPnl: Double
    var dayPnl: Double
    var openPositions: Int
    var closedTrades: Int?
    var winRate: Double
    var maxDrawdown: Double
    var peakCapital: Double?
    var lastUpdated: ActivityTimestamp?
}

File: AalsiTraderAndroid/app/src/main/java/com/aalsitrader/android/model/PaperPortfolio.kt
package com.aalsitrader.android.model

import kotlinx.serialization.Serializable

@Serializable
data class PaperPortfolio(
    val capital: Double = 0.0,
    val startingCapital: Double = 0.0,
    val availableCapital: Double = 0.0,
    val marginUsed: Double = 0.0,
    val totalPnl: Double = 0.0,
    val unrealizedPnl: Double = 0.0,
    val dayPnl: Double = 0.0,
    val openPositions: Int = 0,
    val closedTrades: Int? = null,
    val winRate: Double = 0.0,
    val maxDrawdown: Double = 0.0,
    val peakCapital: Double? = null,
    val lastUpdated: ActivityTimestamp? = null,
)
